test/unit/select_comparators_test.lua

```diff
@@ -139,3 +141,24 @@ g.test_unicode_collations = function()
     t.assert(not func_le_unicode({'A', 'Á', 'Ä'}, {'a', 'A', 'a'}, 3))
     t.assert(func_le_unicode_ci({'A', 'Á', 'Ä'}, {'a', 'A', 'a'}, 3))
 end
+
+g.test_uuid_type_parts = function()
+    local key_parts = {
+        {type = 'uuid'}, {}
+    }
+
+    local func_eq, err = select_comparators.gen_func(operators.EQ, key_parts)
+    t.assert(err == nil)
+    local u1, u2 = uuid.fromstr('8a1ffbc0-241e-11eb-8d27-0800200c9a66'), uuid.fromstr('8f1ffbc0-241e-11eb-8d27-0800200c9a66')
+    local u1_copy = uuid.fromstr(u1:str())
+    t.assert(func_eq({u1, 'str'}, {u1, 'str'}))
+    t.assert(func_eq({u1, 'str'}, {u1_copy, 'str'}))
+    t.assert(not func_eq({u1, 'str'}, {u2, 'str'}))
+
+    local func_ge, err = select_comparators.gen_func(operators.GE, key_parts)
+    t.assert(err == nil)
+    t.assert(not func_ge({u1, 'str'}, {u2, 'str'}))
```

Comment:
This test doesn't looks correct because you assume the lexicographic sorting of strings is the same as for fields of uuid type in the Tarantool core.
Correct way:

```
key_def_lib = require('key_def')
key_def = key_def_lib.new({{field = 1, type = 'uuid'}})
-- It's slow way but we should compare this results with it (at least in tests)
-- because it's a rule how Tarantool sort UUIDs internally
key_def:compare({uuid1}, {uuid2})
```

Reply:
I selected manually two predefined UUID which precede each other by type and by string representation.

Reply:
Splitted into two lines.